provider/datagen/src/provider.rs

```diff
@@ -348,30 +358,51 @@ impl std::fmt::Display for TrieType {
 }
 
 trait IterableDataProviderCached<M: KeyedDataMarker>: DataProvider<M> {
-    fn supported_locales_cached(
+    fn supported_requests_cached(
         &self,
     ) -> Result<HashSet<(DataLocale, DataKeyAttributes)>, DataError>;
 }
 
+impl DatagenProvider {
+    #[allow(clippy::type_complexity)] // not as complex as it appears
+    fn populate<M: KeyedDataMarker>(
```

Comment:
Nit: Since `DatagenProvider` functions are shared across the whole crate, it's better to use more specific names

```suggestion
    fn populate_supported_requests_cache<M: KeyedDataMarker>(
```